client/pluginmanager/csimanager/volume.go

```diff
@@ -61,12 +61,12 @@ func newVolumeManager(logger hclog.Logger, plugin csi.CSIPlugin, rootDir, contai
 	}
 }
 
-func (v *volumeManager) stagingDirForVolume(root string, vol *structs.CSIVolume) string {
-	return filepath.Join(root, StagingDirName, vol.ID, "todo-provide-usage-options")
+func (v *volumeManager) stagingDirForVolume(root string, vol *structs.CSIVolume, usage *UsageOptions) string {
+	return filepath.Join(root, StagingDirName, vol.ID, usage.ToFS())
 }
 
-func (v *volumeManager) allocDirForVolume(root string, vol *structs.CSIVolume, alloc *structs.Allocation) string {
-	return filepath.Join(root, AllocSpecificDirName, alloc.ID, vol.ID, "todo-provide-usage-options")
+func (v *volumeManager) allocDirForVolume(root string, vol *structs.CSIVolume, alloc *structs.Allocation, usage *UsageOptions) string {
+	return filepath.Join(root, AllocSpecificDirName, alloc.ID, vol.ID, usage.ToFS())
```

Comment:
Now that I'm seeing this I'm not totally sure I understand why we'd make the usage options part of the file path. In the commit message you said:

> We pass the attachment/access mode seperately from the volume as it simplifies some handling and doesn't necessarily force every attachment to use the same mode should more be supported (I.e if we let each volume "foo" {} specify an override in the future).

That totally makes sense to me as a parameter to `MountVolume`/`UnmountVolume`, but having this in the file path suggests we can have a volume mounted more than once (in different usage modes) for the same alloc+volume pair?

Reply:
you potentially can, bc as it stands the following is technically valid:

```
volume "foo" {
	type = "csi"
    source = "volume-1"
    readonly = false
}


volume "foo-ro" {
	type = "csi"
    source = "volume-1"
    readonly = true
}
```

But also bc we may want to `NodePublishVolume` for each `volume_mount { }` configuration in the alloc (not sure that we do want to though).

Reply:
Ah gotcha. 👍